AI #1944: Refactor Unit Format (#2131)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/attributes/unit_format.md

  * *FOCUS dataset column* SHOULD use one of the recommended count-based unit names listed below for count-based base units.
  * *FOCUS dataset column* MAY include a count-based base unit that is not listed as one of the allowed values.
  * *FOCUS dataset column* SHOULD use capitalized nouns for base units that do not correspond to any of the allowed base unit names listed below.
* *FOCUS dataset column* MUST use a hyphen ("-") to separate base units when expressing a [compound unit](#attributes.definitions.compoundunit) (e.g., "GB-Hours").
* *FOCUS dataset column* MUST use a slash ("/") to separate the numerator and denominator when expressing a [ratio unit](#attributes.definitions.ratiounit) (e.g., "GB/Hour" to signify gigabytes per hour).
* *FOCUS dataset column* SHOULD use the `<plural-units>` format when expressing a [simple unit](#attributes.definitions.simpleunit) (e.g., "GB", "Seconds").
* *FOCUS dataset column* SHOULD use the `<singular-base-unit>-<plural-base-unit>` format when expressing a compound unit (e.g., "GB-Hours", "MB-Days", "Request-Tokens").

> * *FOCUS dataset column* SHOULD use the `<plural-units>/<singular-time-unit>` format when expressing a ratio unit with a time denominator (e.g., "GB/Hour", "PB/Day").

Reviewer: ```suggestion
* *FOCUS dataset column* SHOULD use the `<plural-units>/<singular-time-unit>` format when expressing a ratio unit with a time denominator (e.g., "GB/Hour", "PB/Day").
* *FOCUS dataset column* SHOULD use the `<compound-unit>/<singular-time-unit>` format when expressing a ratio unit with a compound unit numerator (e.g., "Core-Hours/Day").
```

The Ratio Unit definition at line 51 includes `Core-Hours/Day` as an example, but there's no corresponding SHOULD format for compound-numerator ratio units. This adds explicit format guidance to match.

* *FOCUS dataset column* SHOULD use the `<plural-units>/<denominator-quantity> <plural-time-units>` format when the *FOCUS dataset column* represents a ratio unit and a denominator quantity is included (e.g., "Units/3 Months").

## Definitions

The normative requirements above refer to the following definitions.

### Measurement Unit

A standardized expression that describes how quantities in a *FOCUS dataset* are denominated (e.g., `GB`, `Seconds`, `GB-Hours`, `10 GB/Hour`, `Units/3 Months`).

### Base Unit

An atomic unit that serves as a building block for all [measurement units](#attributes.definitions.measurementunit); can be a data size unit, time-based unit, or count-based unit (e.g., `GB`, `Hour`, `Token`).

### Simple Unit

A [measurement unit](#attributes.definitions.measurementunit) that contains exactly one [base unit](#attributes.definitions.baseunit), optionally preceded by a [unit quantity](#attributes.definitions.unitquantity) (e.g., `GB`, `Seconds`, `1000 Tokens`).

### Compound Unit

> A [measurement unit](#attributes.definitions.measurementunit) that combines two [base units](#attributes.definitions.baseunit) using a hyphen (`-`) to express a quantity sustained over a period, optionally preceded by a [unit quantity](#attributes.definitions.unitquantity) (e.g., `GB-Hours`, `MB-Days`).

Reviewer: ```suggestion
A [measurement unit](#attributes.definitions.measurementunit) that combines two [base units](#attributes.definitions.baseunit) using a hyphen (`-`), optionally preceded by a [unit quantity](#attributes.definitions.unitquantity) (e.g., `GB-Hours`, `MB-Days`, `Request-Tokens`).
```

The format SHOULD at line 22 and `Request-Tokens` example expand compound units beyond time-based combinations. The definition still constrains to "a quantity sustained over a period," which excludes the new non-time use cases. This removes the time constraint and adds `Request-Tokens` to the examples to keep the definition aligned with the format.

If a characterization is preferred over a pure deletion, something like "to express a multi-dimensional quantity" could replace "to express a quantity sustained over a period."

| Data size in bits    | Data size in bytes    |
| :------------------- | :-------------------- |

> | b (bit) = 10^0       | B (byte = 10^0)       |

Reviewer: ```suggestion
| b (bit = 10^0)       | B (byte = 10^0)       |
```

Every other row in the table follows the pattern `abbreviation (name = 10^N)` with the exponent inside the parentheses (e.g., `Kb (kilobit = 10^3)`). The bits column has the closing paren after "bit" with the exponent outside.
